sys/dev/sound/pci/hda/pin_patch_realtek.h

```diff
@@ -0,0 +1,813 @@
+        				{ }
+				}
+			}, {
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(HP_225AID_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_OVERRIDE(0x18, 0x2181103f), /* line in */
+					PIN_OVERRIDE(0x1b, 0x21011020), /* line out */
+        				{ }
+				}
+			}, { }
+		}
+	}, { /**** CODEC: HDA_CODEC_ALC271 ****/
+		.id = HDA_CODEC_ALC271,
+		.patches = (struct model_pin_patch_t[]){
+			{
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(ACER_AO725_SUBVENDOR),
+					PIN_SUBVENDOR(ACER_AO756_SUBVENDOR),
+					PIN_SUBVENDOR(ACER_E1_472_SUBVENDOR),
+					PIN_SUBVENDOR(ACER_E1_572_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_OVERRIDE(0x14, 0x99130110), /* speaker */
+					PIN_OVERRIDE(0x19, 0x01a19c20), /* mic */
+					PIN_OVERRIDE(0x1b, 0x99a7012f), /* int mic */
+					PIN_PATCH_HP_OUT(0x21),
+        				{ }
+				}
+			}, { }
+		}
+	}, { /**** CODEC: HDA_CODEC_ALC280 ****/
+		.id = HDA_CODEC_ALC280,
+		.patches = (struct model_pin_patch_t[]){
+			{
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(HP_2272ID_SUBVENDOR),
+					PIN_SUBVENDOR(HP_2273ID_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_OVERRIDE(0x1b, 0x21011020), /* line out */
+					PIN_PATCH_HPMIC_WITH_DETECT(0x1a),
+					PIN_OVERRIDE(0x18, 0x2181103f), /* line in */
+        				{ }
+				}
+			}, { }
+		}
+	}, { /**** CODEC: HDA_CODEC_ALC288 ****/
+		.id = HDA_CODEC_ALC288,
+		.patches = (struct model_pin_patch_t[]){
+			{
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(DELL_E7240_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = pin_patches_dell_xps_jack
+			}, { }
+		}
+	}, { /**** CODEC: HDA_CODEC_ALC292 ****/
+		.id = HDA_CODEC_ALC292,
+		.patches = (struct model_pin_patch_t[]){
+			{
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(LENOVO_X120BS_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = (struct pin_patch_t[]){
+					PIN_PATCH_STRING(21, "as=1 seq=15"),
+        				{ }
+				}
+			}, { }
+		}
+	}, { /**** CODEC: HDA_CODEC_ALC298 ****/
+		.id = HDA_CODEC_ALC298,
+		.patches = (struct model_pin_patch_t[]){
+			{
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(DELL_XPS9560_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = pin_patches_dell_xps_jack
+			}, { }
+		}
+	}, { /**** CODEC: HDA_CODEC_ALC861 ****/
+		.id = HDA_CODEC_ALC861,
+		.patches = (struct model_pin_patch_t[]){
+			{
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(ASUS_W6F_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = pin_patches_asus_w6f
+			}, {
+				.models = (struct pin_machine_model_t[]){
+					PIN_SUBVENDOR(UNIWILL_9075_SUBVENDOR),
+					{ }
+				},
+				.pin_patches = pin_patches_uniwill_9075
+			}, { }
+		}
+	}, { /**** CODEC: HDA_CODEC_ALC880 ****/
+		.id = HDA_CODEC_ALC880,
+		.patches = (struct model_pin_patch_t[]){
+			{ // old patch
```

Comment:
what are these //old patch comments?

Reply:
Old patches are patches present before I started to work on the hdaa_patches.c file. These included patches, but some in a logical AND and OR way. For all I know, the starting values to apply the AND mask and OR values on can be anything, so for the Realtek chipsets I simply converted the old patches to the structs introduced by me in the ALC github patches so they should keep working. For mask/OR patches and string type patches, I marked the old patches -- already present in the sources when I started working on my patches -- since they were also marked 'old patches' in the old source files. (And were marked in the old source files as candidates to be removed at some point.)